admin/src/main/java/io/strimzi/arguments/topic/DeleteTopicCommand.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.arguments.topic;
+
+import io.strimzi.admin.AdminProperties;
+import org.apache.kafka.clients.admin.Admin;
+import picocli.CommandLine;
+
+@CommandLine.Command(name = "delete")
+public class DeleteTopicCommand extends BasicTopicCommand {
+    @Override
+    public Integer call() {
+        return deleteTopic();
+    }
+
+    private Integer deleteTopic() {
+        Admin admin = Admin.create(AdminProperties.adminProperties(this.bootstrapServer));
```

Comment:
Does this admin create a socket (i.e., TCP connection)? If so, you should also close the connection, because if one executes a lot of `deletetopic()` invocations it would let a lot of ports open I would say. 

If the Admin class implements interface `Closable` you can use such logic with `try with resource` and it will close the connection implicitly for you. Plus apply this logic to all other classes to eliminate It if all mentioned above is true :).

Reply:
yeah makes sense, I forgot to use it, good catch :)